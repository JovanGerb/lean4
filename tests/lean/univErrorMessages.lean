/-!
This file tests errors in the case of uninstantiated universe levels
-/

example : Nat :=
  let x := #[]
  let y := []
  4

example : Nat :=
  let x : Array.{_} _ := #[]
  let y : List.{_+1} _ := []
  4

example : Nat :=
  let x : Array _ := id { data := [] }
  4
