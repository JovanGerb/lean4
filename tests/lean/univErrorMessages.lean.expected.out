univErrorMessages.lean:7:11-7:13: error: don't know how to synthesize implicit argument 'α'
  @List.nil ?m
context:
x : Array ?m := #[]
⊢ Type ?u
univErrorMessages.lean:7:6-7:7: error: failed to infer 'let' declaration type
univErrorMessages.lean:6:11-6:14: error: don't know how to synthesize implicit argument 'α'
  @List.toArray ?m []
context:
⊢ Type ?u
univErrorMessages.lean:6:11-6:14: error: don't know how to synthesize implicit argument 'α'
  @List.nil ?m
context:
⊢ Type ?u
univErrorMessages.lean:6:6-6:7: error: failed to infer 'let' declaration type
univErrorMessages.lean:7:11-7:13: error: don't know how to synthesize universe u of 'List.nil'
  @List.nil.{?u}
univErrorMessages.lean:6:11-6:14: error: don't know how to synthesize universe u of 'List.nil'
  @List.nil.{?u}
univErrorMessages.lean:6:11-6:14: error: don't know how to synthesize universe u_1 of 'List.toArray'
  @List.toArray.{?u}
univErrorMessages.lean:12:26-12:28: error: don't know how to synthesize implicit argument 'α'
  @List.nil ?m
context:
x : Array ?m := #[]
⊢ Type (?u + 1)
univErrorMessages.lean:12:21-12:22: error: don't know how to synthesize placeholder for argument 'α'
context:
x : Array ?m := #[]
⊢ Type (?u + 1)
univErrorMessages.lean:11:25-11:28: error: don't know how to synthesize implicit argument 'α'
  @List.toArray ?m []
context:
⊢ Type ?u
univErrorMessages.lean:11:25-11:28: error: don't know how to synthesize implicit argument 'α'
  @List.nil ?m
context:
⊢ Type ?u
univErrorMessages.lean:11:20-11:21: error: don't know how to synthesize placeholder for argument 'α'
context:
⊢ Type ?u
univErrorMessages.lean:12:26-12:28: error: don't know how to synthesize universe u of 'List.nil'
  @List.nil.{?u + 1}
univErrorMessages.lean:12:16-12:17: error: don't know how to synthesize universe placeholder ?u
univErrorMessages.lean:11:25-11:28: error: don't know how to synthesize universe u of 'List.nil'
  @List.nil.{?u}
univErrorMessages.lean:11:25-11:28: error: don't know how to synthesize universe u_1 of 'List.toArray'
  @List.toArray.{?u}
univErrorMessages.lean:11:17-11:18: error: don't know how to synthesize universe placeholder ?u
univErrorMessages.lean:16:21-16:38: error: don't know how to synthesize implicit argument 'α'
  @id (Array ?m) { data := [] }
context:
⊢ Type ?u
univErrorMessages.lean:16:34-16:36: error: don't know how to synthesize implicit argument 'α'
  @List.nil ?m
context:
⊢ Type ?u
univErrorMessages.lean:16:16-16:17: error: don't know how to synthesize placeholder for argument 'α'
context:
⊢ Type ?u
univErrorMessages.lean:16:34-16:36: error: don't know how to synthesize universe u of 'List.nil'
  @List.nil.{?u}
univErrorMessages.lean:16:21-16:23: error: don't know how to synthesize universe u of 'id'
  @id.{?u + 1}
univErrorMessages.lean:16:10-16:15: error: don't know how to synthesize universe u of 'Array'
  Array.{?u}
